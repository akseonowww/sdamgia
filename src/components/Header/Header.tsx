import React, { FC } from 'react'
import { cn } from '@bem-react/classname'

interface IHeaderProps {
	siteTitle?: string
	className?: string
}

const Header: FC<IHeaderProps> = ({ siteTitle = '' }) => {
	const header = cn('Header')

	return <header className={header()}>Шапка «{siteTitle}»</header>
}

export default Header
