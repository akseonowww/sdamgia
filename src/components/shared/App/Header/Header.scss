@import '../breakpoints';

.Header {
	max-width: 1000px;
	margin: 0 auto;

	&-Logo {
		float: left;
		height: 100px;
		margin: 5px 33px 5px 33px;
	}

	&-Title {
		margin: 0 0 10px 120px;
		padding-top: 5px;

		font-family: Verdana, Geneva, sans-serif;
		font-size: 30px;
		line-height: 30px;
		letter-spacing: 2.5px;
	}

	&-Link {
		text-decoration: none;
		color: #000;

		border-bottom: 0.05rem solid rgba(0, 0, 0, 0.25);

		&:hover {
			color: #f60;

			border-bottom-color: rgba(255, 102, 51, 0.25);
		}

		&_selected {
			color: #000 !important;

			border-bottom: none;
		}
	}
}

.Header .header_site_info {
	margin-left: 120px;
	font-family: Verdana, Geneva, sans-serif;
	font-size: 14px;
	line-height: 20px;
}
.Header .header_subj_name {
	margin-left: 120px;
	font-family: Verdana, Geneva, sans-serif;
	font-size: 20px;
	line-height: 20px;
	margin-top: 15px;
}
@media (max-width: 1000px) {
	.header_subj_logo {
		display: none;
	}
	body {
		margin: 0 5px;
	}
}
@media (max-width: 600px) {
	.header_subj_info {
		display: none;
	}
	.Header img {
		margin: 5px;
		height: 50px;
	}
	.Header .header-title {
		font-size: 16px;
		line-height: 16px;
		margin: 5px;
	}
	.Header .header_site_info {
		margin-left: 60px;
		font-size: 10px;
		line-height: 10px;
	}
	.Header .header_subj_name {
		margin-left: 60px;
		font-size: 14px;
		line-height: 16px;
	}
}
@media (max-width: 360px) {
	.header_site_info {
		display: none;
	}
}
