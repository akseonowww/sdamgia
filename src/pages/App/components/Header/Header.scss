@import 'src/breakpoints';

.Header {
	display: flex;
	justify-content: space-between;
	max-width: 1000px;
	margin: 0 auto;

	&-Main {
		flex-grow: 1;
	}

	&-Logo {
		float: left;
		margin: 5px;
		width: 50px;
		height: 50px;

		@include media-breakpoint-up(sm) {
			margin: 5px 33px 5px 33px;
			width: 100px;
			height: 100px;
		}
	}

	&-Title {
		margin: 5px;
		padding-top: 5px;

		font-size: 16px;
		line-height: 16px;
		letter-spacing: 2.5px;

		@include media-breakpoint-up(sm) {
			margin: 0 0 10px;

			font-size: 30px;
			line-height: 30px;
		}
	}

	&-SubjectImg {
		display: none;
		width: 346.667px;
		height: 100px;
		margin: 5px;

		@include media-breakpoint-up(mil) {
			display: block;
		}
	}

	&-WebsiteDesc {
		display: none;

		@include media-breakpoint-up(sm) {
			display: block;

			font-size: 14px;
			line-height: 20px;
		}
	}

	&-SubjectTitle {
		font-size: 14px;
		line-height: 16px;

		@include media-breakpoint-up(sm) {
			margin-top: 15px;

			font-size: 20px;
			line-height: 20px;
		}
	}
}
