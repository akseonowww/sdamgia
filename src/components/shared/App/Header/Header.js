import React from 'react';
import { NavLink } from 'react-router-dom';

import './Header.scss';

const Header = ({ className }) => (
	<header className={`Header ${className}`}>
		<NavLink to="/" exact activeClassName="Link_selected">
			<img
				src="https://ege.sdamgia.ru/img/headers/logo.svg"
				className="Header-Logo"
				alt="Сдам ГИА"
			/>
		</NavLink>
		<img
			className="header_subj_logo"
			src="https://math.reshuct.by/img/headers/math.png"
			style={{
				float: 'right',
				height: '100px',
				margin: '5px'
			}}
			alt="Математика"
		/>
		<div className="Header-Title">
			<a
				className="Header-Link"
				target="_blank"
				rel="noopener noreferrer"
				href="https://sdamgia.ru"
			>
				СДАМ ГИА
			</a>
			:{' '}
			<NavLink
				to="/"
				exact
				className="Header-Link"
				activeClassName="Header-Link_selected Link_selected"
			>
				РЕШУ ЕГЭ
			</NavLink>
		</div>
		<div className="header_site_info">
			Образовательный портал для подготовки к экзаменам
		</div>
		<div className="header_subj_name">Математика профильного уровня</div>
		<div style={{ clear: 'both' }}></div>
	</header>
);

export default Header;
