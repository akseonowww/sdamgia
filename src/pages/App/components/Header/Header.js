import React from 'react';
import { NavLink } from 'react-router-dom';
import cx from 'classnames';

import 'components/Link/Link.scss';
import 'components/Link/_selected/Link_selected.scss';
import './Header.scss';

const Header = ({ className }) => (
	<header className={cx('Header', className)}>
		<div className="Header-Main">
			<NavLink activeClassName="Link_selected" to="/" exact>
				<img
					className="Header-Logo"
					src="https://ege.sdamgia.ru/img/headers/logo.svg"
					alt="Сдам ГИА"
				/>
			</NavLink>

			<div className="Header-Title">
				<a
					className="Header-Link Link Link_black"
					href="https://sdamgia.ru"
					rel="noopener noreferrer"
				>
					СДАМ ГИА
				</a>
				:{' '}
				<NavLink
					className="Header-Link Link Link_black"
					activeClassName="Link_selected"
					to="/"
					exact
				>
					РЕШУ ЕГЭ
				</NavLink>
			</div>
			<div className="Header-WebsiteDesc">
				Образовательный портал для подготовки к экзаменам
			</div>
			<div className="Header-SubjectTitle">
				Математика профильного уровня
			</div>
		</div>

		<img
			className="Header-SubjectImg"
			src="https://math.reshuct.by/img/headers/math.png"
			alt="Математика"
		/>
	</header>
);

export default Header;
