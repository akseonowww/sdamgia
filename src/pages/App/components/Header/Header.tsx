import React, { FC } from 'react'
import cx from 'classnames'

import Link from 'components/Link'
import './Header.scss'

interface IHeaderProps {
  className?: string
}

const Header: FC<IHeaderProps> = ({ className }) => (
  <header className={cx('Header', className)}>
    <div className="Header-Main">
      <Link activeClassName to="/" exact>
        <img
          className="Header-Logo"
          src="https://ege.sdamgia.ru/img/headers/logo.svg"
          alt="Сдам ГИА"
        />
      </Link>

      <div className="Header-Title">
        <Link className="Link_black Header-Link" href="https://sdamgia.ru">
          СДАМ ГИА
        </Link>
        :{' '}
        <Link className="Link_black Header-Link" activeClassName to="/" exact>
          РЕШУ ЕГЭ
        </Link>
      </div>
      <div className="Header-WebsiteDesc">
        Образовательный портал для подготовки к экзаменам
      </div>
      <div className="Header-SubjectTitle">Математика профильного уровня</div>
    </div>

    <img
      className="Header-SubjectImg"
      src="https://math.reshuct.by/img/headers/math.png"
      alt="Математика"
    />
  </header>
)

export default Header
