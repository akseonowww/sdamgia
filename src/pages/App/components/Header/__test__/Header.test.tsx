import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'
import { MemoryRouter } from 'react-router-dom'

import Header from '../Header'

describe('Header', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
        container
      )
    })
    expect(container && container.textContent).toBe(
      'СДАМ ГИА: РЕШУ ЕГЭОбразовательный портал для подготовки к экзаменамМатематика профильного уровня'
    )

    act(() => {
      render(
        <MemoryRouter>
          <Header className="PageLayout-Header" />
        </MemoryRouter>,
        container
      )
    })
    expect(container && container.textContent).toBe(
      'СДАМ ГИА: РЕШУ ЕГЭОбразовательный портал для подготовки к экзаменамМатематика профильного уровня'
    )
  })

  it('should render a content', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <MemoryRouter>
          <Header className="PageLayout-Header" />
        </MemoryRouter>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
