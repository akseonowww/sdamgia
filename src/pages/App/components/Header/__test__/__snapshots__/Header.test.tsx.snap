// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a content 1`] = `
"<header class=\\"Header\\">
  <div class=\\"Header-Main\\"><a aria-current=\\"page\\" class=\\"Link Link_selected\\" href=\\"/\\"><img class=\\"Header-Logo\\" src=\\"https://ege.sdamgia.ru/img/headers/logo.svg\\" alt=\\"Сдам ГИА\\"></a>
    <div class=\\"Header-Title\\"><a class=\\"Link Link_black Header-Link\\" href=\\"https://sdamgia.ru\\" rel=\\"noopener noreferrer\\">СДАМ ГИА</a>: <a aria-current=\\"page\\" class=\\"Link Link_black Header-Link Link_selected\\" href=\\"/\\">РЕШУ ЕГЭ</a></div>
    <div class=\\"Header-WebsiteDesc\\">Образовательный портал для подготовки к&nbsp;экзаменам</div>
    <div class=\\"Header-SubjectTitle\\">Математика профильного уровня</div>
  </div><img class=\\"Header-SubjectImg\\" src=\\"https://math.reshuct.by/img/headers/math.png\\" alt=\\"Математика\\">
</header>"
`;

exports[`Header should render a content 2`] = `
"<header class=\\"Header PageLayout-Header\\">
  <div class=\\"Header-Main\\"><a aria-current=\\"page\\" class=\\"Link Link_selected\\" href=\\"/\\"><img class=\\"Header-Logo\\" src=\\"https://ege.sdamgia.ru/img/headers/logo.svg\\" alt=\\"Сдам ГИА\\"></a>
    <div class=\\"Header-Title\\"><a class=\\"Link Link_black Header-Link\\" href=\\"https://sdamgia.ru\\" rel=\\"noopener noreferrer\\">СДАМ ГИА</a>: <a aria-current=\\"page\\" class=\\"Link Link_black Header-Link Link_selected\\" href=\\"/\\">РЕШУ ЕГЭ</a></div>
    <div class=\\"Header-WebsiteDesc\\">Образовательный портал для подготовки к&nbsp;экзаменам</div>
    <div class=\\"Header-SubjectTitle\\">Математика профильного уровня</div>
  </div><img class=\\"Header-SubjectImg\\" src=\\"https://math.reshuct.by/img/headers/math.png\\" alt=\\"Математика\\">
</header>"
`;
